storage/remote/queue_manager.go

```diff
@@ -907,53 +1012,70 @@ func (s *shards) sendSamplesWithBackoff(ctx context.Context, samples []prompb.Ti
 		return nil
 	}
 
+	retry := func() {
+		s.qm.metrics.retriedSamplesTotal.Add(float64(sampleCount))
+	}
+
+	err = sendWriteRequestWithBackoff(ctx, s.qm.cfg, s.qm.client(), s.qm.logger, req, attemptStore, retry)
+	if err != nil {
+		return err
+	}
+	s.qm.metrics.succeededSamplesTotal.Add(float64(sampleCount))
+	s.qm.metrics.bytesSent.Add(float64(reqSize))
+	s.qm.metrics.highestSentTimestamp.Set(float64(highest / 1000))
+	return nil
+}
+
+func sendWriteRequestWithBackoff(ctx context.Context, cfg config.QueueConfig, s WriteClient, l log.Logger, req []byte, attempt func(int) error, retry func()) error {
```

Comment:
The variable name `retry` makes me think that the function being passed would do a full retry. Perhaps a name like `onRetry()` to make it clear that it is more of an event callback?